lib/topic_query.rb

```diff
@@ -339,6 +343,13 @@ def self.unread_filter(list, staff: false)
                regular: TopicUser.notification_levels[:regular], tracking: TopicUser.notification_levels[:tracking])
   end
 
+  def self.unseen_filter(list, user_first_seen_at, staff)
```

Comment:
Does this need to be declared on `self` when it's only being used in the same class and nowhere else?

Reply:
There are several other filters next to this one:
- `self.new_filter`
- `self.unread_filter`
- `self.tracked_filter`

Some of them are in use outside of the class. Looks like it's better to stick to the same pattern here, even though we don't use unseen_filter somewhere else yet.

Reply:
> even though we don't use `unseen_filter` somewhere else yet.

In this case, I think it'll be better if we keep the method private since there is no need for it to be public. If we need it to be public in the future, we can easily make it so.

Reply:
IMO in this specific case sticking to the pattern is better, because it makes code more consistent (even taking into account that some plugin can start using it, and we'll need to support backward compatibility). 

But ok, making it private will work, so I've changed it.